Content.Server/Atmos/Reactions/N2ODecompositionReaction.cs

```diff
@@ -0,0 +1,27 @@
+using Content.Server.Atmos.EntitySystems;
+using Content.Shared.Atmos;
+using JetBrains.Annotations;
+
+namespace Content.Server.Atmos.Reactions;
+
+/// <summary>
+///     Decomposes Nitrous Oxide into Nitrogen and Oxygen.
+/// </summary>
+[UsedImplicitly]
+public sealed partial class N2ODecompositionReaction : IGasReactionEffect
+{
+    public ReactionResult React(GasMixture mixture, IGasMixtureHolder? holder, AtmosphereSystem atmosphereSystem, float heatScale)
+    {
+        var cacheN2O = mixture.GetMoles(Gas.NitrousOxide);
+
+        var burnedFuel = cacheN2O * Atmospherics.N2ODecompositionRate * (mixture.Temperature - Atmospherics.N2ODecompositionMinScaleTemp) * (mixture.Temperature - Atmospherics.N2ODecompositionMaxScaleTemp) / Atmospherics.N2ODecompositionScaleDivisor;
```

Comment:
Why is this a quadratic function? The numbers chosen for `N2ODecompositionMinScaleTemp` and `N2ODecompositionMaxScaleTemp` are also so large that this quadratic is effectively linear in any reasonable temperature range. Could you please talk a bit more about how you came up with this equation and why the numbers are the way they are?

Reply:
i didn't came up with it. as i said i violently copied the code from /tg/station 🤷‍♂️